expression/builtin_string.go

```diff
@@ -922,9 +922,36 @@ func (c *convertFunctionClass) getFunction(ctx sessionctx.Context, args []Expres
 		return nil, err
 	}
 	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, types.ETString, types.ETString)
-	// TODO: issue #4436: The second parameter should be a constant.
-	// TODO: issue #4474: Charset supported by TiDB and MySQL is not the same.
```

Comment:
#4474 contains two cases:

> 1. Charsets supported by TiDB may not supported in MySQL, like "866"
> 2. Charsets supported by MySQL may not supported in TiDB, like "binary"

Case 2 is fixed by supporting binary in TiDB, and case 1 about "866" is covered in the modification of test case [here](https://github.com/pingcap/tidb/pull/9305/files#diff-dd33813624d57c13ba9794ebc82005c4R896).

In this PR, I think a charset can be converted to should satisfy 2 constrains:
1. This charset has a transformer defined in `charset.encodings` of **parser/charset/encoding_table.go**
2. This charset has a default collation defined in `charset.charsetInfos` of **parser/charset/charset.go**（or we don't know which collation it will be converted to）.

For charset "866", it does not have a default collation, so I think it is "not supported by TiDB", meeting the expected behavior in #4474:
```
MySQL > select convert("123" using "866");
ERROR 1115 (42000): Unknown character set: '866'
```